src/components/permalink/PermalinkService.js

```diff
@@ -82,6 +82,26 @@ goog.require('ga_urlutils_service');
             return baseEmbed + '?' + gaUrlUtils.toKeyValue(newParams);
           };
 
+          this.getImgHref = function(p) {
+            var newParams = angular.extend({}, params);
+            if (angular.isDefined(p)) {
+              angular.extend(newParams, p);
+            }
+            if (angular.isDefined(newParams.mobile)) {
+              delete newParams.mobile;
+            }
+            var baseImg = base.replace(/^http:/, 'https:').
+                replace(/(index|mobile)\.html$/, 'img.html');
+            if (!/img\.html$/.test(baseImg)) {
+              if (!/\/$/.test(baseImg)) {
+                baseImg += '/';
+              }
+              baseImg += 'img.html';
+            }
+            return baseImg + '?' + gaUrlUtils.toKeyValue(newParams);
+          };
```

Comment:
Lot of duplicate code with getEmbedHref.

Can't we use the same function for both of them with a page name in parameter ?